src/input/mod.rs

```diff
@@ -129,32 +127,42 @@ impl<'a> InputFile<'a> {
     }
 }
 
-impl<'a> Drop for InputFile<'a> {
-    fn drop(&mut self) {
-        unsafe { CEXR_InputFile_delete(self.handle) };
-    }
-}
-
-struct IStream<'a> {
-    handle: *mut CEXR_IStream,
-    _phantom: PhantomData<&'a ()>,
-}
+impl<'a> InputFile<'a, &'a [u8]> {
+    pub fn from_slice<'b>(slice: &'b [u8]) -> Result<InputFile<'b, &'b [u8]>> {
+        let istream_ptr = unsafe {
```

Comment:
Hmm... possibly.  It would make `T` unsized in this case, but maybe that's okay.  I'll give it a try.